rdrf/rdrf/email_notification.py

```diff
@@ -17,55 +16,111 @@ class RdrfEmail(object):
 
     _DEFAULT_LANGUAGE = "en"
 
-    def __init__(self, reg_code=None, description=None, language="en", email_notification=None):
+    def __init__(self, reg_code=None, description=None, email_notification=None, language=None):
         self.email_from = None
-        self.recipient = []
+        self.recipients = []
         self.email_templates = []
-        self.language = language
         self.template_data = {}
 
         self.reg_code = reg_code
         self.description = description
+        self.language = language # used to only send to subset of languages by EmailNotificationHistory resend
 
-        self.email_notification = email_notification
+        if email_notification:
+            self.email_notification = email_notification
+            self.reg_code = self.email_notification.registry.code
+            self.description = self.email_notification.description
+            
+        else:
+            self.email_notification = self._get_email_notification()
+            
 
     def send(self):
```

Comment:
yeah for "Accept-Language: en-US,en;q=0.5" string I think the code will split on the hyphen and take the first member of the resulting list  ( from memory - I will test it )